infra/modules/gcp/main.tf

```diff
@@ -26,6 +27,36 @@ resource "google_project_service" "gcp_infra_api" {
   disable_on_destroy         = false # disabling on destroy has potential to conflict with other uses of the project
 }
 
+resource "google_artifact_registry_repository" "psoxy-functions-repo" {
+  location      = var.bucket_location
+  project       = var.project_id
+  repository_id = "psoxy-functions"
+  description   = "Docker repository used on the cloud functions"
+  format        = "DOCKER"
+
+  ## Not supported in current google providers, needs 5.14 as there it is GA
+  # See https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#5140-jan-29-2024
+  # but even is present in the documentation (https://registry.terraform.io/providers/hashicorp/google/4.80.0/docs/resources/artifact_registry_repository#argument-reference)
+  # when applied it throws an error with the message: "An argument named "cleanup_policy_dry_run" is not expected here"
+  # and "no block for cleanup_policies" is expected
+  /*cleanup_policy_dry_run = false
+
+  # https://cloud.google.com/artifact-registry/docs/repositories/cleanup-policy#json_2
+  # https://registry.terraform.io/providers/hashicorp/google/4.80.0/docs/resources/artifact_registry_repository#argument-reference
+  cleanup_policies {
```

Comment:
... so mainly for this. Adding a `cleanup_policy` we could just put rules for keeping the most recent images instead of storing all of them.

But it doesn't work with 4.84.0, 5.0.0 and it will only work since 5.14.0, which I have to modify the versioning of the gcp providers (from 3.74 to 5.14). Not sure if we can do it, but if so, it could be interesting. In any case, I've left the commented code here, at some point we could use it once we update the versions

Reply:
Leaving this commented with a TODO